Shared/Coordinator/Coordinator.swift

```diff
@@ -102,9 +102,7 @@ class Coordinator<Result>: NSObject, CoordinatorType {
         self.removeFromParent()
         self.onFinishedFlow?(result)
         
-        Task {
-            await self.taskPool.cancelAndRemoveAll()
-        }
+        self.taskPool.cancelAndRemoveAll()
```

Comment:
I'm making task pool a class which makes it much easier to use. No need to cancel it in a Task. This should lead to more predictable behavior.